static/src/javascripts-legacy/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -160,14 +182,91 @@ define([
                     trackDisplay: true,
                     cssModifierClass: colourClass
                 }).show(renderedBanner);
+
+            if(params.secondaryButtonCaption &&  SECONDARY_BUTTON != null) {
+                setSecondaryButtonListener();
+            }
+
             if (messageShown) {
                 mediator.emit('membership-message:display');
             }
             mediator.emit('banner-message:complete');
         }
 
+        function emailIsValid(email){
+            return typeof email === 'string' && email.indexOf('@') > -1;
+        }
+
+        function sendEmail(email){
+            submitForm(email, LIST_ID).then(showThankYouMessage, showErrorMessage)
+        }
+
+        function showThankYouMessage(){
+            hideElement(REMIND_ME_TEXT_FIELD);
+            hideElement(REMIND_ME_CTA);
+            hideElement(REMIND_ME_ERROR);
+
+            showElement(REMIND_ME_THANKS_MESSAGE);
+        }
+
+        function showErrorMessage(){
+            hideElement(REMIND_ME_TEXT_FIELD);
+            hideElement(REMIND_ME_CTA);
+            showElement(REMIND_ME_ERROR);
+        }
+
+        function submitForm(email, listID) {
+            var formQueryString =
+                'email=' + encodeURI(email) + '&' +
+                'listId=' + listID;
+
+            return fetch(config.page.ajaxUrl + EMAIL_PATH,
+                        {   method: 'post',
+                            body: formQueryString,
+                            headers: {
+                                'Accept': 'application/json'
+                            }
+                        }
+            );
+        }
+
+        function getDOMElements(){
+            REMIND_ME_FORM = document.querySelector(REMIND_ME_FORM_SELECTOR);
+            REMIND_ME_TEXT_FIELD = document.querySelector(REMIND_ME_TEXT_FIELD_SELECTOR);
```

Comment:
Preference: Since this variable is only used once, could we just pass the string directly in here?

Or declare the variable within the`getDOMElements` function? That way I don't have to scroll back up to remind myself which selector is being used:

`document.querySelector('.membership__engagement-text-field');`